virttest/libvirt_xml/vm_xml.py

```diff
@@ -633,7 +633,13 @@ def _cleanup(details=""):
         if not vmxml.undefine():
             _cleanup(details="Undefine VM %s failed" % vm.name)
         # Alter the XML
+        str_old = "domain-" + vm.name
+        str_new = "domain-" + new_name
         vmxml.vm_name = new_name
+        for channel in vmxml.get_agent_channels():
+            for child in channel._children:
```

Comment:
Yes, we reviewed that bug yesterday, and there's some things need update here.
But I just check function 'get_agent_channels()' and it return an agent name list, so I guess it's not right to use it in this way.
And, function 'set_agent_channel' also need update.